### ISSUE TITLE
MOTS - Missions 6/9 fail to load, Missions 11/13 spawn the player OOB, and 12 crashes to desktop

### ISSUE THREAD

**Issue by JordanRadar** [shinyquagsire23/OpenJKDF2#132]
I have played through all of MOTS that I am currently able on the most recent version of OpenJKDF2 (0.6.15) and I do have some things to report. (I realize you stated that the game was impossible to complete in the latest release, but it didn't stop me from trying. Maybe this information will be useful to you?)

Mission 1 - Appears to work fine, no issues discovered while playtesting.
Mission 2 - Appears to work fine, no issues discovered while playtesting.
Mission 3 - Appears to work fine, no issues discovered while playtesting.
Mission 4 - Appears to work fine, no issues discovered while playtesting.
Mission 5 - Appears to work fine, no issues discovered while playtesting.
Mission 6 - Ingame message comes up reporting that level failed to load. No crash.
Mission 7 - Appears to work fine, no issues discovered while playtesting.
Mission 8 - Appears to work fine, no issues discovered while playtesting.
Mission 9 - Ingame message comes up reporting that level failed to load. No crash.
Mission 10 - Appears to work fine, no issues discovered while playtesting.
Mission 11 - Spawns player OOB (See Screenshot Below)
![image](https://user-images.githubusercontent.com/82469061/205469163-605b380d-0d34-47fd-9739-d852140e3f7f.png)
Mission 12 - Crashes to desktop.
Mission 13 - Spawns player OOB (See Screenshot Below)
![image](https://user-images.githubusercontent.com/82469061/205469194-f04e6ecd-9db3-4cd3-839c-cd303b80500c.png)
Mission 14 - Appears to work fine, no issues discovered while playtesting (have not completed the level as of this moment).

In Summary:
Missions 1-5, 7-8, and 10 work. 6 fails to load, 9 fails to load, 11 spawns you OOB with no way to clip back into bounds, and 12 straight up crashes the game. 13 also appears to load you out of bounds with no way to get back in, and 14 seems to work fine although I haven't gone through the whole mission.

**Comment by CansecoDev** [comment]
For the multiplayer maps, Spaceport seems to be the only one that crashes.

**Comment by katastic** [comment]
I'm up to mission 13, no unplayable problems. Crashes on occasion, but everything works, no unbeatable levels yet.

Windows 10 64-bit, Geforce GTX 1070, Intel i7 cpu

**Comment by shinyquagsire23** [comment]
Super helpful, thanks for detailing all of that. I'll start going through the ones that crash first since those are probably (hopefully lmao) easy fixes compared to the OOB ones.

**Comment by shinyquagsire23** [comment]
Fixed the load failures with 8b6c39bddb097552c421c7f688febc9eb38ce0a5, to be included in the next release. It also seems to have fixed the OOB spawn issues which is interesting, but it does make sense. The parser was only getting the first 1024 characters of each line in the JKL.

Working on 12 now because that one appears to be a separate issue.

**Comment by shinyquagsire23** [comment]
Should be fixed [v0.6.16](https://github.com/shinyquagsire23/OpenJKDF2/releases/tag/v0.6.16), but I'll hold off on closing just in case

**Comment by shinyquagsire23** [comment]
@CansecoDev I can confirm that Spaceport is also working now

**Comment by shinyquagsire23** [comment]
Played through all the levels myself, safe to say it's all fixed I think
